src/varpubs/pubmed_db.py

```diff
@@ -14,24 +24,171 @@ class PubmedArticle(SQLModel, table=True):
     doi: str
 
 
+class TermToPMID(SQLModel, table=True):
+    term: str = Field(nullable=False)
+    pmid: int = Field(Integer, nullable=False)
+    gene: str = Field(nullable=False)
+
+    __table_args__ = (sqlalchemy.PrimaryKeyConstraint("term", "pmid", "gene"),)
+
+
 @dataclass
 class PubmedDB:
     path: Path
-    _engine: sqlalchemy.engine.base.Engine | None = field(init=False)
+    vcf_paths: Iterable[Path]
+    email: str
+    _engine: Optional[sqlalchemy.engine.base.Engine] = field(init=False, default=None)
 
     def deploy(self) -> None:
-        # wipe existing database
-        if self.path.exists():
-            self.path.unlink()
-        # create parent directories if they don't exist
+        logger.info(f"Deploying database at: {self.path}")
+        logger.info(f"Input VCF paths: {[str(p) for p in self.vcf_paths]}")
+        logger.info(f"Entrez email: {self.email}")
+
+        self.create_tables()
+        terms = self.extract_terms()
+        Entrez.email = self.email
+
+        with Session(self.engine) as session:
+            all_pmids = set()
+            term_to_pmids = {}
+
+            for term in terms:
+                try:
+                    handle = Entrez.esearch(db="pubmed", term=term, retmax=3)
+                    record = cast(dict, Entrez.read(handle))
+                    pmids = list(map(int, record.get("IdList", [])))
+                    if pmids:
+                        match = re.match(r"(\w+)\sp\.", term)
+                        gene = match.group(1) if match else "UNKNOWN"
+                        term_to_pmids[term] = [(pmid, gene) for pmid in pmids]
+                        all_pmids.update(pmids)
+                except Exception as e:
+                    logger.error(f"Error fetching PMIDs for term '{term}': {e}")
+
+            # Fetch metadata in batch
+            self.fetch_articles_metadata(session, list(all_pmids))
+
+            # Store term→pmid mappings
+            for term, pmid_gene_list in term_to_pmids.items():
+                for pmid, gene in pmid_gene_list:
+                    exists = session.exec(
+                        select(TermToPMID).where(
+                            TermToPMID.term == term,
+                            TermToPMID.pmid == pmid,
+                            TermToPMID.gene == gene,
+                        )
+                    ).first()
+                    if not exists:
+                        session.add(TermToPMID(term=term, pmid=pmid, gene=gene))
+
+            session.commit()
+            logger.info("Data committed to the database.")
+
+    def create_tables(self) -> None:
         if self.path.parent != Path():
             self.path.parent.mkdir(parents=True, exist_ok=True)
```

Comment:
```suggestion
self.path.parent.mkdir(parents=True, exist_ok=True)
```

I think this check is not needed with `exist_ok=True` right?